crates/prune/src/parts/mod.rs

```diff
@@ -0,0 +1,123 @@
+mod receipts;
+pub(crate) use receipts::Receipts;
+
+use crate::PrunerError;
+use reth_db::database::Database;
+use reth_interfaces::RethResult;
+use reth_primitives::{BlockNumber, PruneCheckpoint, PruneMode, PrunePart, TxNumber};
+use reth_provider::{
+    BlockReader, DatabaseProviderRW, PruneCheckpointReader, PruneCheckpointWriter,
+};
+use std::ops::RangeInclusive;
+use tracing::error;
+
+/// A part represents a segment of data for pruning.
+///
+/// Parts are called from [Pruner](crate::Pruner) with the following lifecycle:
+/// 1. Call [Part::prune] with `delete_limit` of [PruneInput].
+/// 2. If [Part::prune] returned a [Some] in `checkpoint` of [PruneOutput], call
+///    [Part::save_checkpoint].
+/// 3. Subtract `pruned` of [PruneOutput] from `delete_limit` of next [PruneInput].
+pub(crate) trait Part {
```

Comment:
can we think of another name than just Part?

having a hard time with the `Part` terminology.

it's "part" of the pruning process, hence the name?

maybe PruneTarget, PruneStep, something like that?

Reply:
renamed to `Segment`